tests/C-tests/threshold_matrix_typed.c

```diff
@@ -75,7 +77,11 @@ int TYPED_FUNC(
     A = bml_random_matrix(matrix_type, matrix_precision, N, M, distrib_mode);
     REAL_T scale_factor = 1.e-2;
     bml_scale_inplace(&scale_factor, A);
-    bml_print_bml_matrix(A, 0, max_row, 0, max_col);
+    if (bml_getMyRank() == 0)
+    {
+        printf("A scaled by %f \n", scale_factor);
+        bml_print_bml_matrix(A, 0, max_row, 0, max_col);
```

Comment:
This function needs to be called from all MPI tasks, not just task 0. This is probably why the MPI tests hang

Reply:
Thanks. Hopefully I caught all instances.